core/CronArchive/QueueConsumer.php

```diff
@@ -311,6 +315,66 @@ public function getNextArchivesToProcess()
         return $archivesToProcess;
     }
 
+    // public for tests
+    public function repairInvalidationsIfNeeded($archiveToProcess)
+    {
+        $table = Common::prefixTable('archive_invalidations');
+
+        $bind = [
+            $archiveToProcess['idsite'],
+            $archiveToProcess['name'],
+            $archiveToProcess['period'],
+            $archiveToProcess['date1'],
+            $archiveToProcess['date2'],
+        ];
+
+        $reportClause = '';
+        if (!empty($archiveToProcess['report'])) {
+            $reportClause = " AND report = ?";
+            $bind[] = $archiveToProcess['report'];
+        }
+
+        $sql = "SELECT DISTINCT period FROM `$table` WHERE idsite = ? AND name = ? AND period > ? AND ? >= date1 AND date2 >= ? $reportClause";
```

Comment:
btw is the distinct actually needed?

Reply:
could do an array_unique as well, was trying to reduce the amount of data selected since there could potentially be a lot of invalidations for a site

Reply:
all good should be fine to keep distinct for now